src/System.CommandLine.Tests/GetValueByNameParserTests.cs

```diff
@@ -194,6 +194,48 @@ public void When_an_option_and_argument_use_same_name_on_the_same_level_of_the_t
             .Where(ex => ex.Message == $"Command {command.Name} has more than one child named \"{sameName}\".");
     }
 
+    [Fact]
+    public void When_options_use_same_name_on_different_levels_of_the_tree_no_exception_is_thrown()
+    {
+        const string sameName = "same";
+
+        RootCommand command = new()
+        {
+            new Command("left")
+            {
+                new Option<int>(sameName)
+            },
+            new Command("right")
+            {
+                new Option<int>(sameName)
+            },
+        };
+
+        command.Parse($"left {sameName} 1").GetValue<int>(sameName).Should().Be(1);
+        command.Parse($"right {sameName} 2").GetValue<int>(sameName).Should().Be(2);
+    }
+
+    [Fact]
+    public void When_the_same_option_used_in_different_levels_of_the_tree_no_exception_is_thrown()
```

Comment:
This is a test for a bug I've hit while working on https://github.com/dotnet/sdk/pull/49181:

```log
System.InvalidOperationException: Command build has more than one child named "--interactive".
```